src/member-api.js

```diff
@@ -200,16 +205,28 @@ export class MemberApi extends TypedEmitter {
         case InviteResponse_Decision.DECISION_UNSPECIFIED:
           return InviteResponse_Decision.REJECT
         case InviteResponse_Decision.ACCEPT:
-          await this.#rpc.sendProjectJoinDetails(deviceId, {
-            inviteId,
-            projectKey: this.#projectKey,
-            encryptionKeys: this.#encryptionKeys,
-          })
+          try {
+            await this.#rpc.sendProjectJoinDetails(deviceId, {
+              inviteId,
+              projectKey: this.#projectKey,
+              encryptionKeys: this.#encryptionKeys,
+            })
+          } catch {
+            throw new ProjectDetailsSendFail()
+          }
 
           // Only add after we know they got the details
           // Otherwise the joiner will be stuck unable to join
           await this.#roles.assignRole(deviceId, roleId)
 
+          try {
+            await this.#waitForInitialSyncWithPeer(deviceId, abortSignal)
+          } catch {
+            // Mark them as "left" so we can retry the flow
+            await this.#roles.assignRole(deviceId, LEFT_ROLE_ID)
```

Comment:
just noting that I really don't have the best idea about whether this introduces any unintended side effects. think it should be okay but just wanted to be honest about it 😄